From eac86715bec987401c229dfecce148302adfa235 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Wed, 13 Mar 2024 15:18:36 +0000
Subject: [PATCH] WIP: split args

---
 src/main.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main.rs b/src/main.rs
index 0761c03..db052d5 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -47,7 +47,8 @@ fn fallible_main() -> Result<bool> {
         no_binaries && no_exported_libraries
     };
     if let Some(target) = args.rust_target {
-        args.forwarded_args.push(format!("--target {}", target).into());
+        args.forwarded_args.push("--target".into());
+        args.forwarded_args.push(target.into());
     };
 
     let verb = if is_pure_library { "check" } else { "build" };
